### PR TITLE
Testing: add vote unit test and coverage script

### PR DESCRIPTION
**Pull request by KeyserSosa**
This adds a rudimentary "votes work as expected" unit test as well as a script to test code coverage using `nosetests` and the `coverage` package which has been added to `setup.py`

:eyeglasses: @Deimos @spladug

### CODE REVIEW

File: r2/setup.py [hunk 1]
```diff
@@ -89,7 +89,7 @@
     # * https://pythonhosted.org/setuptools/setuptools.html#declaring-extras-optional-features-with-their-own-dependencies
     # * https://github.com/pypa/sampleproject/blob/300f04dc44df51492deb859ac98ba521d2c7a17a/setup.py#L71-L77
     extras_require={
-        'test': ['mock', 'nose'],
+        'test': ['mock', 'nose', 'coverage'],
```
Comment (spladug) [inline r2/setup.py]: huh, what about putting these in `tests_require` instead of an extras?

File: r2/coverage.sh [hunk 2]
```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+set -e
+
+BASEDIR=$(dirname $0)
+cd $BASEDIR
+
+VERSION=$(git rev-parse HEAD)
+COVERDIR="/tmp/cover-$VERSION"
```
Comment (spladug) [inline r2/coverage.sh]: how about putting this in `build/` so it doesn't rely on system paths?

**Comment by KeyserSosa** [comment]
:haircut: 

@spladug `python setup.py test` now works with those `setup.py` changes but I had to make some changes to `app_globals.py` which was doing some (rather infuriating to track down) mocking based on a perceived "environment".

**Comment by KeyserSosa** [comment]
:eyeglasses: @Deimos @spladug 

Apparently harold didn't pick up my ninja-edit of the reviewers list.  Being all explicit-like.

**Comment by spladug** [comment]
:fish:

**Comment by KeyserSosa** [comment]
merged in af49fa322bd018dbe69a89a75f4e22fbef6521b3

**Pull request closed without merging**